File: ProgrammePython/fonctionsTableaux.py

# Fonction pour afficher le tableau des intervalles de balayage
def afficher_tableau_balayage(tableau_intervals):
    # Affiche le tableau des intervalles de balayage.
    # Chaque ligne correspond à un intervalle exploré et affiche les informations associées.
    print("\nTableau des intervalles de balayage :")
    print("{:<20} {:<18} {:<18} {:<18} {:<18}".format('Intervalle', 'Borne Inf.', 'Borne Sup.', 'f(Borne Inf.)', 'f(Borne Sup.)'))
    print("-" * 90)

    for interval_data in tableau_intervals:
        print("{:<20} {:<18} {:<18} {:<18} {:<18}".format(
            interval_data['intervalle'],
            interval_data['borne_inferieure'],
            interval_data['borne_superieure'],
            interval_data['f(borne_inferieure)'],
            interval_data['f(borne_superieure)']
        ))


# Fonction pour afficher le tableau stylisé
def afficher_tableau_dichotomie(tableau):
    # Affichage de l'entête du tableau avec des bornes d'étoiles
    print("*" * 100)
    print(f"{'Iteration':<12}|{'Borne Inférieure':<20}|{'Borne Supérieure':<20}|{'f(Borne Inférieure)':<25}|{'f(Borne Supérieure)':<25}|{'f(borne inferieure) * f(borne superieure)':<25}")
    print("*" * 100)

    # Affichage des informations ligne par ligne
    for ligne in tableau:
        print(f"{ligne['iteration']:<12}{ligne['borne_inferieure']:<20}{ligne['borne_superieure']:<20}{ligne['signe_f(borne_inferieure)']:<25}{ligne['signe_f(borne_superieure)']:<25}{ligne['signe_(f(borne_inferieure)*f(borne_superieure))']:<25}")
    
    # Ligne de fermeture avec des étoiles
    print("*" * 100)


def afficher_tableau_secante(tableau_interactions):
    # Affiche le tableau des itérations de manière lisible.
    # Chaque ligne correspond à une itération et affiche les informations associées.

    print("\nTableau des itérations :")
    
    # Définition des en-têtes de colonnes
    en_tetes = ['Iteration', 'Borne Inf.', 'Borne Sup.', 'Estimation', 'f(Estimation)', 'Diff. Bornes']
    
    # Format de l'en-tête pour l'affichage
    print("{:<12} {:<18} {:<18} {:<18} {:<18} {:<18}".format(*en_tetes))
    
    # Ligne de séparation
    print("-" * 90)

    # Parcours des données dans le tableau des itérations
    for iteration_data in tableau_interactions:
        iteration_num = iteration_data['iteration']
        borne_inferieure = iteration_data['borne_inferieure']
        borne_superieure = iteration_data['borne_superieure']
        estimation = iteration_data['estimation']
        f_estimation = iteration_data['f(estimation)']
        diff_bornes = iteration_data['différence_bornes']
        
        # Formatage et affichage des données de l'itération
        print("{:<12} {:<18} {:<18} {:<18} {:<18} {:<18}".format(
            iteration_num,
            borne_inferieure,
            borne_superieure,
            estimation,
            f_estimation,
            diff_bornes
        ))


# Fonction pour afficher le tableau du point fixe
def afficher_tableau_point_fixe(tableau_interactions):
    """
    Affiche le tableau des itérations de manière lisible.
    Chaque ligne correspond à une itération et affiche les informations associées.
    """
    print("\nTableau des itérations :")
    print("{:<12} {:<18} {:<18} {:<18}".format('Iteration', 'x', 'phi(x)', 'Différence'))
    print("-" * 70)

    for iteration_data in tableau_interactions:
        iteration_num = iteration_data['iteration']
        x = iteration_data['x']
        phi_x = iteration_data['phi(x)']
        difference = iteration_data['différence']
        
        # Affichage des données de l'itération
        print("{:<12} {:<18} {:<18} {:<18}".format(
            iteration_num,
            x,
            phi_x,
            difference
        ))

    

File: Programmephp/includes/fonctionsTableaux.php
<?php

// Fonction pour afficher le tableau des intervalles de balayage
function afficher_tableau_balayage($tableau_intervals) {
    echo "\nTableau des intervalles de balayage :\n";
    echo str_pad("Intervalle", 20) . str_pad("Borne Inf.", 18) . str_pad("Borne Sup.", 18) . str_pad("f(Borne Inf.)", 18) . str_pad("f(Borne Sup.)", 18) . "\n";
    echo str_repeat("-", 90) . "\n";

    foreach ($tableau_intervals as $interval_data) {
        echo str_pad($interval_data['intervalle'], 20) .
             str_pad($interval_data['borne_inferieure'], 18) .
             str_pad($interval_data['borne_superieure'], 18) .
             str_pad($interval_data['f(borne_inferieure)'], 18) .
             str_pad($interval_data['f(borne_superieure)'], 18) . "\n";
    }
}

// Fonction pour afficher le tableau stylisé pour la dichotomie
function afficher_tableau_dichotomie($tableau) {
    echo str_repeat("*", 100) . "\n";
    echo str_pad("Iteration", 12) . "|" . str_pad("Borne Inférieure", 20) . "|" . str_pad("Borne Supérieure", 20) . "|" .
         str_pad("f(Borne Inférieure)", 25) . "|" . str_pad("f(Borne Supérieure)", 25) . "|" .
         str_pad("f(Borne Inf.) * f(Borne Sup.)", 25) . "\n";
    echo str_repeat("*", 100) . "\n";

    foreach ($tableau as $ligne) {
        echo str_pad($ligne['iteration'], 12) .
             str_pad($ligne['borne_inferieure'], 20) .
             str_pad($ligne['borne_superieure'], 20) .
             str_pad($ligne['signe_f(borne_inferieure)'], 25) .
             str_pad($ligne['signe_f(borne_superieure)'], 25) .
             str_pad($ligne['signe_(f(borne_inferieure)*f(borne_superieure))'], 25) . "\n";
    }

    echo str_repeat("*", 100) . "\n";
}

// Fonction pour afficher le tableau de la méthode de la sécante
function afficher_tableau_secante($tableau_interactions) {
    echo "\nTableau des itérations :\n";

    // En-têtes des colonnes
    $en_tetes = ['Iteration', 'Borne Inf.', 'Borne Sup.', 'Estimation', 'f(Estimation)', 'Diff. Bornes'];
    echo str_pad($en_tetes[0], 12) . str_pad($en_tetes[1], 18) . str_pad($en_tetes[2], 18) .
         str_pad($en_tetes[3], 18) . str_pad($en_tetes[4], 18) . str_pad($en_tetes[5], 18) . "\n";
    echo str_repeat("-", 90) . "\n";

    foreach ($tableau_interactions as $iteration_data) {
        echo str_pad($iteration_data['iteration'], 12) .
             str_pad($iteration_data['borne_inferieure'], 18) .
             str_pad($iteration_data['borne_superieure'], 18) .
             str_pad($iteration_data['estimation'], 18) .
             str_pad($iteration_data['f(estimation)'], 18) .
             str_pad($iteration_data['différence_bornes'], 18) . "\n";
    }
}

// Fonction pour afficher le tableau du point fixe
function afficher_tableau_point_fixe($tableau_interactions) {
    echo "\nTableau des itérations :\n";
    echo str_pad("Iteration", 12) . str_pad("x", 18) . str_pad("phi(x)", 18) . str_pad("Différence", 18) . "\n";
    echo str_repeat("-", 70) . "\n";

    foreach ($tableau_interactions as $iteration_data) {
        echo str_pad($iteration_data['iteration'], 12) .
             str_pad($iteration_data['x'], 18) .
             str_pad($iteration_data['phi(x)'], 18) .
             str_pad($iteration_data['différence'], 18) . "\n";
    }
}
